language/functional_tests/tests/testsuite/generics/call.mvir

```diff
@@ -0,0 +1,39 @@
+//! no-run: verifier, runtime
+
+module M {
+    resource Foo {}
+
+    public id<T>(x: T) {
+        return move(x);
+    }
+
+    public bar<T1, T2>(x: T1, y: T2) {
+    }
+
+    foo() {
+        let x: u64;
+        let y: Self.Foo;
+
+        x = Self.id<u64>(3);
+        y = Self.id<Self.Foo>(Foo {});
+        Self.bar<Self.Foo, bool>();
+    }
+}
+
+// The following checks & nots ensures the locals_signatures table has at least 6 elements
+//     and the first 6 elements match the exact pattern we define.
+// This is to ensure we can safely refer to elements of the table by indices.
+
+// check: locals_signatures
+// not: LocalsSignature
+// check: LocalsSignature
+// not: LocalsSignature
+// check: LocalsSignature
+// not: LocalsSignature
+// check: LocalsSignature
+// not: LocalsSignature
+// check: LocalsSignature([U64]), LocalsSignature([Struct(StructHandleIndex(0), [])]), LocalsSignature([Struct(StructHandleIndex(0), []), Bool])
+
+// check: Call(0, LocalsSignatureIndex(3))
```

Comment:
Do we want to write checks that depend on the structure of the AST like this in the functional tests? It seems a bit brittle. The `check`/`not` pattern used here is particularly hard to understand.

Feel free to defer the tests to a separate PR if my test bikeshedding is slowing you down from landing the meat of the changes :).

Reply:
Good point. TLDR: my plan is to check these tests in for now, but remove them once generics is fully implemented.

These tests form our only line of defense for now, and I don't think it is a good idea to check in hundreds of lines of unchecked code. From my opinion, it is okay or even beneficial that these tests are a little bit brittle: generics are under heavy development and I'd like to be extra-sensitive to anything that would change them during this time.

Once the verifier & runtime are implemented and we write many many more tests, we will be confident enough to remove these checks, and rely on the code going through the whole pipeline successfully for correctness, just like how other tests work.

Reply:
Ok, that strategy makes sense to me. Thanks!